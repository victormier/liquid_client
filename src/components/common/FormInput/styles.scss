@import 'base/variables';
@import 'base/mixins';

.input {
  @include roundedBlock;
  @include transparentBlueBackground;
  margin-bottom: $base-grid-gutter/4;
  color: white;

  &::placeholder {
    color: white;
  }

  &:disabled {
    background: transparent;
    border: 1px solid transparentize(white, 0.5);
  }
}
