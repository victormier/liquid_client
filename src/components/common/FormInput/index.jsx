import React from 'react';
import styles from './styles.scss';

const FormInput = props => (
  <input
    {...props}
    className={styles.input}
  />
);

export default FormInput;
