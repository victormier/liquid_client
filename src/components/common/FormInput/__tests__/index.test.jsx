import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import FormInput from '../index';


describe('component', () => {
  it('renders', () => {
    const container = shallow(
      <FormInput
        type="text"
        value="test me"
        placeHolder="test"
      />
    );

    expect(shallowToJson(container)).toMatchSnapshot();
  });
});
