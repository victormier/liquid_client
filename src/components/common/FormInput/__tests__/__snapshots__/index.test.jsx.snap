// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<input
  placeHolder="test"
  type="text"
  value="test me"
/>
`;
